internal/services/controllers/v1/olap/process_dag_status_updates.go

```diff
@@ -68,50 +83,54 @@ func (o *OLAPControllerImpl) updateDAGStatuses(ctx context.Context, tenantId str
 	}
 
 	if o.prometheusMetricsEnabled {
-		workflowNames, err := o.repo.Workflows().ListWorkflowNamesByIds(ctx, tenantId, workflowIds)
-		if err != nil {
-			return false, err
-		}
+		for _, tenantId := range tenantIds {
```

Comment:
This isn't great, but I figured it's about equivalent to what we have now processing tenants one at a time anyways